src/formatting_stack/linters/kondo.clj

```diff
@@ -11,22 +11,15 @@
 
 (def default-options
   {:cache     true
-   :cache-dir formatting-stack.kondo-classpath-cache/cache-dir
+   :parallel  true
    :linters   {:cond-else            off ;; undesired
                :missing-docstring    off ;; undesired
                :unused-binding       off ;; undesired
                :private-call         off ;; undesired
                :unresolved-symbol    off ;; can give false positives
                :unused-symbol        off ;; can give false positives
-               :unused-private-var   off ;; can give false positives
```

Comment:
formatting-stack was designed with a specific philosophy, namely to be integrated with REPLs, and gathering runtime info from the Clojure compiler and built-time introspection system, and from precise `eval`-based tools such as Eastwood.

`The formatters are run before the linters` is not quite a complete assessment because some of these faults can't be possibly fixed by a formatter. Part of the reason why these kondo linters are disabled is because they are already covered precisely by either the Clojure compiler (e.g. `unresolved-var`) or Eastwood.

Another reason is the occasional inaccuracy of clj-kondo's approach (otherwise very valuable and which I do imitate from time to time). Inaccurate warnings hinder a repl-friendly experience that can also be integrated with CI checks.

I'd much appreciate these removals were proposed and discussed in a per-case basis in an issue.

Reply:
> formatting-stack was designed with a specific philosophy, namely to be integrated with REPLs, and gathering runtime info from the Clojure compiler and built-time introspection system, and from precise eval-based tools such as Eastwood.

Eastwood is not guaranteed to run and thus the code isn't guaranteed to be loaded. Especially when configuration becomes more flexible it's plausible that users will opt-out of Eastwood in REPL-use because of the latency.

> some of these faults can't be possibly fixed by a formatter.

That's why I removed these _defaults_. I think if clj-kondo comes across these issues they should be reported. Apparently we couldn't fix them, or the formatters didn't run.

> Part of the reason why these kondo linters are disabled is because they are already covered precisely by either the Clojure compiler or Eastwood

The duplicated linters are still disabled, the clj-kondo warning seems friendlier than the Clojure reader exception. Happy to disable the `unresolved-var` linter if you feel strongly about it; but it doesn't seem like a big deal.

> Another reason is the occasional inaccuracy of clj-kondo [...]

`:unused-private-var` is actually unintended removal in this case, shouldn't have been removed. 

The intended change was just about the linters backed by formatters.

Reply:
> Eastwood is not guaranteed to run and thus the code isn't guaranteed to be loaded.

Crafting a custom 'stack' should also involve a sensible set of defaults that work holistically (i.e. no duplicates, and no false negatives either).

f-s offers a number of stack which all work holistically very accurately.

When crafting a custom stack that e.g. omitted Eastwood, it would be certainly OK to use different clj-kondo defaults as you suggest. But worsening f-s defaults for a _half-configured stack_ seems misguided to me.

Also, please observe that f-s has a diverse user base and accordingly we shouldn't make breaking changes (and preferrably also avoid changes in philosophy - people have shown interest in f-s as designed with a specific philosophy - mutating it to something else might be confusing)

Remember that CI can be set up depending on f-s, hence my extra caution.

As a constructive suggestion, I'd like to point out that a refactored config system (e.g. https://github.com/nedap/formatting-stack/pull/152 or an alternative design) could easily make this problem more elegant and less delicate to tackle.

I would encourauge  you to keep exploring config mechanisms, and see how much simplification can we squeeze out of it.

A specific tricky/valuable part is making our clj-kondo linter detect that eastwood is present, and auto-shrink its own config accordingly (probably this is easier to fix at an upper layer, e.g. strategies).